SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialSender.h

```diff
@@ -124,6 +125,20 @@ class SPATIALGDK_API USpatialSender : public UObject
 	bool ValidateOrExit_IsSupportedClass(const FString& PathName);
 
 private:
+	struct EntityComponentsDeleter
+	{
+		void operator()(TArray<FWorkerComponentData>* Components) const noexcept;
+	};
+
+	using EntityComponents = TUniquePtr<TArray<FWorkerComponentData>, EntityComponentsDeleter>;
```

Comment:
Alternative implementation for consideration - don't have a unique ptr, copy the array to each delegate, then call a function to delete the components in the array on success or non-timeout failure.

Alternative that I prefer is to wrap each schema object in a unique ptr that has a deleter and only have arrays of that type. However this is a bigger thing that would introduce inconsistencies in the codebase unless rolled out more generally.

Reply:
This is actually what is in now. After moving stuff to the entity factory it made it harder to read if the different ways of creating entities did too different things.